pkg/lock/lock.go

```diff
@@ -19,10 +19,16 @@
 package lock
 
 import (
+	"errors"
 	"os"
 	"sync"
 )
 
+var (
+	// ErrLocked is returned if the underlying fd is already locked.
+	ErrLocked = errors.New("lock: file already locked")
```

Comment:
I think prefix `lock:` is not a good idea. the consumer will add any prefix for this error.